### ISSUE TITLE
Unknown lvalue 'Capab ilityBoundSet' in section 'Service'

### ISSUE THREAD

**Issue by keddad** [yggdrasil-network/yggdrasil-go#600]
Running Debian with the latest version of yggdrasil, it fails to start with the following logs:

```
root@vps708527:~# sudo systemctl status yggdrasil
● yggdrasil.service - yggdrasil
   Loaded: loaded (/etc/systemd/system/yggdrasil.service; enabled; vendor preset
: enabled)
   Active: failed (Result: exit-code) since Fri 2019-11-01 07:58:21 C
ET; 9min ago
 Main PID: 4266 (code=exited, status=2)
      CPU: 19ms

Nov 01 07:58:21 vps708527 systemd[1]: yggdrasil.service: Unit entered fa
iled state.
Nov 01 07:58:21 vps708527 systemd[1]: yggdrasil.service: Failed with res
ult 'exit-code'.
Nov 01 08:01:16 vps708527 systemd[1]: [/etc/systemd/system/yggdrasil.ser
vice:11] Unknown lvalue 'CapabilityBoundSet' in section 'Service'
Nov 01 08:01:45 vps708527 systemd[1]: [/etc/systemd/system/yggdrasil.ser
vice:11] Unknown lvalue 'CapabilityBoundSet' in section 'Service'
Nov 01 08:02:02 vps708527 systemd[1]: [/etc/systemd/system/yggdrasil.ser
vice:11] Unknown lvalue 'CapabilityBoundSet' in section 'Service'
Nov 01 08:02:33 vps708527 systemd[1]: [/etc/systemd/system/yggdrasil.ser
vice:11] Unknown lvalue 'CapabilityBoundSet' in section 'Service'
Nov 01 08:02:34 vps708527 systemd[1]: [/etc/systemd/system/yggdrasil.ser
vice:11] Unknown lvalue 'CapabilityBoundSet' in section 'Service'
Nov 01 08:03:12 vps708527 systemd[1]: [/etc/systemd/system/yggdrasil.ser
vice:11] Unknown lvalue 'CapabilityBoundSet' in section 'Service'
Nov 01 08:03:15 vps708527 systemd[1]: [/etc/systemd/system/yggdrasil.service:11] Unknown lvalue 'Capab
ilityBoundSet' in section 'Service'
Nov 01 08:03:15 vps708527 systemd[1]: [/etc/systemd/system/yggdrasil.service:11] Unknown lvalue 'Capab
ilityBoundSet' in section 'Service'

```
Since the only changes I made to initial config were "Listen" and "Peers", I suppose it is some kind of bug

**Comment by keddad** [comment]
Looks like I can't even restart it
```
root@vps708527:~# sudo systemctl restart yggdrasil
root@vps708527:~# sudo systemctl status yggdrasil
● yggdrasil.service - yggdrasil
   Loaded: loaded (/etc/systemd/system/yggdrasil.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Fri 2019-11-01 08:13:23 CET; 182ms ago
  Process: 14484 ExecStart=/usr/bin/yggdrasil -useconffile /etc/yggdrasil.conf (code=exited, status=2)
  Process: 14481 ExecStartPre=/bin/sh -ec if ! test -s /etc/yggdrasil.conf;                  then umask 077;                  yggdrasil -genconf > /etc/yggdrasil.co
nf;                  echo 'WARNING: A new /etc/yggdrasil.conf file has been generated.';              fi (code=exited, status=0/SUCCESS)
  Process: 14480 ExecStartPre=/sbin/modprobe tun (code=exited, status=0/SUCCESS)
 Main PID: 14484 (code=exited, status=2)
      CPU: 20ms

Nov 01 08:13:22 vps708527 systemd[1]: yggdrasil.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
Nov 01 08:13:22 vps708527 systemd[1]: yggdrasil.service: Unit entered failed state.
Nov 01 08:13:22 vps708527 systemd[1]: yggdrasil.service: Failed with result 'exit-code'.
Nov 01 08:13:23 vps708527 systemd[1]: yggdrasil.service: Service hold-off time over, scheduling restart.
Nov 01 08:13:23 vps708527 systemd[1]: Stopped yggdrasil.
Nov 01 08:13:23 vps708527 systemd[1]: yggdrasil.service: Start request repeated too quickly.
Nov 01 08:13:23 vps708527 systemd[1]: Failed to start yggdrasil.
Nov 01 08:13:23 vps708527 systemd[1]: yggdrasil.service: Unit entered failed state.
Nov 01 08:13:23 vps708527 systemd[1]: yggdrasil.service: Failed with result 'exit-code'.

```

**Comment by neilalexander** [comment]
Which Debian version?

Please remove the `CapabilityBoundSet` line from `/etc/systemd/system/yggdrasil.service`, run `systemctl daemon-reload` and try again.

**Comment by keddad** [comment]
> Which Debian version?
> 
> Please remove the `CapabilityBoundSet` line from `/etc/systemd/system/yggdrasil.service`, run `systemctl daemon-reload` and try again.

Debian 9.11, after removing this line my logs have a log of messages like this one:
```
Nov 01 08:22:06 vps708527 systemd[1]: yggdrasil.service: Service hold-off time over, scheduling restart.
Nov 01 08:22:06 vps708527 systemd[1]: Stopped yggdrasil.
Nov 01 08:22:06 vps708527 systemd[1]: Starting yggdrasil...
Nov 01 08:22:06 vps708527 systemd[1]: Started yggdrasil.
Nov 01 08:22:06 vps708527 yggdrasil[864]: panic: Found a punctuator character '}' when expecting a quoteless string (check your syntax) at line 158,1 >>> }
Nov 01 08:22:06 vps708527 yggdrasil[864]: goroutine 1 [running]:
Nov 01 08:22:06 vps708527 yggdrasil[864]: main.readConfig(0xc00001672d, 0xc000046b60, 0xc00001672e, 0x2)
Nov 01 08:22:06 vps708527 yggdrasil[864]:         /home/circleci/project/cmd/yggdrasil/main.go:77 +0xa23
Nov 01 08:22:06 vps708527 yggdrasil[864]: main.main()
Nov 01 08:22:06 vps708527 yggdrasil[864]:         /home/circleci/project/cmd/yggdrasil/main.go:160 +0x1e90
Nov 01 08:22:06 vps708527 systemd[1]: yggdrasil.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
Nov 01 08:22:06 vps708527 systemd[1]: yggdrasil.service: Unit entered failed state.
Nov 01 08:22:06 vps708527 systemd[1]: yggdrasil.service: Failed with result 'exit-code'.
Nov 01 08:22:07 vps708527 systemd[1]: yggdrasil.service: Service hold-off time over, scheduling restart.
Nov 01 08:22:07 vps708527 systemd[1]: Stopped yggdrasil.
Nov 01 08:22:07 vps708527 systemd[1]: yggdrasil.service: Start request repeated too quickly.
Nov 01 08:22:07 vps708527 systemd[1]: Failed to start yggdrasil.
Nov 01 08:22:07 vps708527 systemd[1]: yggdrasil.service: Unit entered failed state.
Nov 01 08:22:07 vps708527 systemd[1]: yggdrasil.service: Failed with result 'exit-code'.
```

**Comment by neilalexander** [comment]
That one looks like something has been entered wrong into the configuration file. 

Run `yggdrasil -useconffile /etc/yggdrasil.conf -normaliseconf` - this will print out a “cleaned” version of the config and should show you what isn’t being parsed properly. You can then fix `/etc/yggdrasil.conf`.

**Comment by neilalexander** [comment]
You can also try `dpkg-reconfigure yggdrasil` to try repairing `/etc/yggdrasil.conf`.

**Comment by keddad** [comment]
> You can also try `dpkg-reconfigure yggdrasil` to try repairing `/etc/yggdrasil.conf`.

Ok, I found the issue. But it seems weird. Config says that
```

  # Listen addresses for incoming connections. You will need to add
  # listeners in order to accept incoming peerings from non-local nodes.
  # Multicast peer discovery will work regardless of any listeners set
  # here. Each listener should be specified in URI format as above, e.g.
  # tcp://0.0.0.0:0 or tcp://[::]:0 to listen on all interfaces.
```

But if the listener is specified as in example:
`Listen: [ tcp://0.0.0.0:0 ]`
It fails to load. If it is empty:
`Listen: []`
It works fine. Appears to be some kind of bug, after all.

**Comment by whoizit** [comment]
not works:
```
Listen: [ tcp://0.0.0.0:0 ]
```

works:
```
Listen: [ tcp://0.0.0.0:0 
]
```

**Comment by neilalexander** [comment]
The HJSON parser library we use is a little bit picky - the `Listen` lines really should be on their own lines, like so:
```
Listen: [
  tcp://0.0.0.0:0
]
```
... or on one line, in quotes:
```
Listen: [ "tcp://0.0.0.0:0" ]
```

**Comment by Arano-kai** [comment]
>```
> Nov 01 08:03:12 vps708527 systemd[1]: [/etc/systemd/system/yggdrasil.ser
> vice:11] Unknown lvalue 'CapabilityBoundSet' in section 'Service'
> ```

Fixed in #597 PR

**Comment by neilalexander** [comment]
This is in v0.3.12, which is due to be merged as per #610. Closing this issue as a result.
